### ISSUE TITLE
brainstorming/analogy - painting process

### ISSUE THREAD

**Issue by dobkeratops** [bbernhard/imagemonkey-core#156]
Basically annotation is a **painting** task.

The recent tweaks have made many cases much easier (hence see the improved activity chart :) ),

I can explain clearer now what was wrong originally, and where to go to improve further:-

**the original workflow is like being told to paint images one colour at a time** 
"here's a red brush.. here's a random image.. draw **all** the red. now here's a green brush.. here's another random image.. draw **all** the green..' <etc>

Now with the browser mode, at least you pick one and find the images where a chosen brush works ok (hence success with road, pavement - clear areas).

The "add labels" mode works ok because it's more like asking _"what palette does this image need?"_ .. that's doable.

But the best way to handle an image is to see the context and gradually refine it.. (i) pick the predominant areas, block them out ("start with the mid-tones and broad brush shapes"), then refine by clipping details out.(add the highlights and shadows.. refine the boundaries..). the optimum order depends on the size and specific shapes (zoomed in, you clip out what isn't the subject.. but zoomed out, you start with the background then draw the subjects)

Saying "the unpainted area is a certain set of labels" might work ok, because this is like a background fill. Hence the idea of an 'unoccluded sky' label, clipping out the patches of sky between vegetation..  and the idea "consider the un-labelled area as a mix of each un-used label"

Telling what colours are is actually difficult in the real world because of relative perception.. labels aren't far off, because many shapes in the distance aren't clear, and there's overlapping categories.. but sometimes once you figure out surrounding context, those difficult parts get easier.

From this POV i'll re-iterate the idea of **general label blending** #141: you do get areas where it's optimum to mix '2 colours' (2 labels).. this is especially true of any organic parts (e.g. mixture of soil, bushes, grass, fallen leaves,bark,moss) but also anything moving. You sometimes get vegetation growing between paving stones (so a label "pavement/grass" would be optimal).

Then there's bits of road covered in gravel around a construction site ("gravel/road"), moving crowds, trash or litter (cleaning bot..)

<img width="865" alt="screen shot 2018-06-27 at 06 59 57" src="https://user-images.githubusercontent.com/1120754/41955428-e7e6c4f4-79d7-11e8-9ec3-b038612724a2.png">
that's a nice example of complex ground. you might start with a blend 'concrete/soil/grass/gravel'  to fill the floor,("fill with a grey/green/brown random texture") then clip out the areas which are clearly one ('the clear patches of grass, clear patches of concrete, etc'). as I'm looking at it now i'm still finding it hard to guess what the distant areas are (dry soil, or gravel, or dirty concrete?). It looks mostly concrete.. but the presence of grass implies other parts must be soil..

**Comment by dobkeratops** [comment]
heres another experiment in GIMP, different workflow idea..
(i) draw major boundaries in a layer
(ii) flood-fill between them 

there's no occlusion and it's limited to how ever many colours you can easily distinguish, e.g 7 ish (unless you split into more layers, in which case you might aswell go back to channels)

it's easy to draw the boundaries but you occasionally miss parts
you can set up the layer opacity to see what you're doing
it is easier to flick between a few categories in the colour palette

**Comment by bbernhard** [comment]
Wow, many thanks for the feedback - very much appreciated! I need to think a bit more about how we can add the concept of label blending to the current system, but I can definitely see it's advantages. 

What I am wondering is: Do we need to define the blent-together labels already in the labels view or can we leave the decision whether labels will be blent up to the annotator? 

Currently, we have a pretty straight forward concept of: label first, then annotate. I think for "normal" labels that decision makes sense, but for blent together labels I am not really sure. 

I guess sometimes it's up to the annotator if he wants to combine labels or not (sometimes you have to combine, as there is no clear distinction between objects). And as the labeling guy and the annotator aren't necessarily the same, I am wondering if it makes sense to define the label blending already on label level. Or is it better to stick with base labels and let the annotator do the label blending? 

I know, that this doesn't fit 100% in the task based approach (with the clear separation of concerns) that we have right now, but I _think_ this could be a more natural approach.

**Comment by dobkeratops** [comment]
> Do we need to define the blent-together labels already in the labels view or can we leave the decision whether labels will be blent together up to the annotator?

i've added quite a few foo/bar examples in 'add labels' where i could see it would make sense - but at the same time, I think sometimes it would be great when annotating to see related labels (and existing annotations). like with 'person,man,woman' .. toggling between those to annotate all the people would probably be handy

it might just be enough to let the user know the related labels exist (and ideally see their annotations in a feinter shade perhaps) for context ("ok i'll annotate grass, but i know this mixed area can be done as a different ```grass/soil``` task later")

**Comment by dobkeratops** [comment]
>> it probably also depends a lot on which device you are working 

definitely.. the difference between using a  laptop, and using a laptop+wacom is massive.

having said that, even with the wacom many blended areas remain (e.g. the seperate parts being few pixels. this can be either in the distance, or in intricate surfaces
